exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1419 @@
+	case asset.Futures:
+		myPairs, err := ku.GetFuturesOpenContracts(ctx)
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, len(myPairs))
+		for x := range myPairs {
+			pairs[x], err = currency.NewPairFromString(strings.ToUpper(myPairs[x].Symbol))
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pairs, nil
+	case asset.Spot, asset.Margin:
+		myPairs, err := ku.GetSymbols(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+		pairs := make(currency.Pairs, len(myPairs))
+		for x := range myPairs {
+			pairs[x], err = currency.NewPairFromString(strings.ToUpper(myPairs[x].Name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		return pairs, nil
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, assetType)
+	}
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores
+// them in the exchanges config
+func (ku *Kucoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assets := ku.GetAssetTypes(false)
+	for a := range assets {
+		// Since the tradable pairs of Margin is same as that of Spot, calling the tradable pairs for Margin is not necessary.
+		// Instead, calling the UpdatePairs() method with Spot and Margin for Spot instruments saves call to FetchTradablePair() method with Margin.
+		if assets[a] == asset.Margin {
+			continue
+		}
+		pairs, err := ku.FetchTradablePairs(ctx, assets[a])
+		if err != nil {
+			return err
+		}
+		err = ku.UpdatePairs(pairs, assets[a], false, forceUpdate)
+		if err != nil {
+			return err
+		}
+		if assets[a] == asset.Spot {
+			err = ku.UpdatePairs(pairs, asset.Margin, false, forceUpdate)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (ku *Kucoin) UpdateTicker(ctx context.Context, p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	p, err := ku.FormatExchangeCurrency(p, assetType)
+	if err != nil {
+		return nil, err
+	}
+	if err := ku.UpdateTickers(ctx, assetType); err != nil {
+		return nil, err
+	}
+	return ticker.GetTicker(ku.Name, p, assetType)
+}
+
+// UpdateTickers updates all currency pairs of a given asset type
+func (ku *Kucoin) UpdateTickers(ctx context.Context, assetType asset.Item) error {
+	switch assetType {
+	case asset.Futures:
+		ticks, err := ku.GetFuturesOpenContracts(ctx)
+		if err != nil {
+			return err
+		}
+		for x := range ticks {
+			pair, err := currency.NewPairFromString(ticks[x].Symbol)
+			if err != nil {
+				return err
+			}
+			err = ticker.ProcessTicker(&ticker.Price{
+				Last:         ticks[x].LastTradePrice,
+				High:         ticks[x].HighPrice,
+				Low:          ticks[x].LowPrice,
+				Volume:       ticks[x].VolumeOf24h,
+				Ask:          ticks[x].IndexPrice,
+				Bid:          ticks[x].MarkPrice,
+				Pair:         pair,
+				ExchangeName: ku.Name,
+				AssetType:    assetType,
+			})
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	case asset.Spot:
```

Comment:
I think it would actually be better and more in line with our other work such as Binance to make this
```suggestion
	case asset.Spot, asset.Margin:
```

Yes, it adds a duplicate entry, but it is handled better. Your current assumption + comment (Which I know I did suggest 😆) means if a user _only_ enables Margin pairs and _not_ spot, then they will not get ticker information. This is highlighted in exchange_wrapper_issues:
![image](https://user-images.githubusercontent.com/9261323/216202178-414e8fc4-124f-43e1-834b-4b0684849ee5.png)